src/cpp/src/generation_config.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <fstream>
+#include <limits>
+
+#include <nlohmann/json.hpp>
+#include <openvino/runtime/core.hpp>
+#include "openvino/genai/generation_config.hpp"
+#include "utils.hpp"
+
+
+namespace ov {
+namespace genai {
+
+GenerationConfig::GenerationConfig(const std::string& json_path) {
+    using ov::genai::utils::read_json_param;
+
+    std::ifstream f(json_path);
+    OPENVINO_ASSERT(f.is_open(), "Failed to open '" + json_path + "' with generation config");
+
+    nlohmann::json data = nlohmann::json::parse(f);
+    
+    read_json_param(data, "max_new_tokens", max_new_tokens);
+    read_json_param(data, "max_length", max_length);
+    // note that ignore_eos is not present in HF GenerationConfig
+    read_json_param(data, "num_beam_groups", num_beam_groups);
+    read_json_param(data, "num_beams", num_beams);
+    read_json_param(data, "diversity_penalty", diversity_penalty);
+    read_json_param(data, "length_penalty", length_penalty);
+    read_json_param(data, "num_return_sequences", num_return_sequences);
+    read_json_param(data, "no_repeat_ngram_size", no_repeat_ngram_size);
+    read_json_param(data, "temperature", temperature);
+    read_json_param(data, "top_p", top_p);
+    read_json_param(data, "top_k", top_k);
+    read_json_param(data, "do_sample", do_sample);
+    read_json_param(data, "repetition_penalty", repetition_penalty);
+    read_json_param(data, "eos_token_id", eos_token_id);
```

Comment:
@pavel-esir 
we agreed to read it from `tokenizer_config.json`, `config.json` in case it's absent in `generation_config.json`

Reply:
ok, I see it's  overridden later in generate method / LLMPipeline ctor